switches/http/http.cpp

```diff
@@ -0,0 +1,1280 @@
+
+/*
+ asebahttp - a switch to bridge HTTP to Aseba
+ 2014-12-01 David James Sherman <david dot sherman at inria dot fr>
+ 
+ Provide a simple REST interface with introspection for Aseba devices.
+ 
+    GET  /nodes                                 - JSON list of all known nodes
+    GET  /nodes/:NODENAME                       - JSON attributes for :NODENAME
```

Comment:
That's a common placeholder notation used in routing definitions (Ruby, Mojolicious). The other convention is {NODENAME}; would that be clearer? I want the reader to understand to show that pattern like `/nodes/:NODENAME/:EVENT/:PARAM1/:PARAM2` (or `/nodes/{NODENAME}/{EVENT}/{PARAM1}/{PARAM2}`) could be instantiated with a URL `http://localhost:3000/nodes/thymio-II/record_sound/4/60`.